salt/modules/keystone.py

```diff
@@ -287,38 +323,44 @@ def endpoint_list(profile=None, **connection_args):
     '''
     kstone = auth(profile, **connection_args)
     ret = {}
+
     for endpoint in kstone.endpoints.list():
-        ret[endpoint.id] = {'id': endpoint.id,
-                            'region': endpoint.region,
-                            'adminurl': endpoint.adminurl,
-                            'internalurl': endpoint.internalurl,
-                            'publicurl': endpoint.publicurl,
-                            'service_id': endpoint.service_id}
+        ret[endpoint.id] = dict((value, getattr(endpoint, value)) for value in dir(endpoint)
+                                if not value.startswith('_') and
+                                isinstance(getattr(endpoint, value), (unicode, dict, bool)))
```

Comment:
This is not Python 3 compatible. Please change all the references from `unicode` to `six.text_type`. Thanks.